goship.go

```diff
@@ -836,20 +904,52 @@ func checkAuth(fn http.HandlerFunc, a auth) http.HandlerFunc {
 	})
 }
 
-// remove projects where user has no access
-func cleanProjects(cp []goship.Project, r *http.Request, u User) []goship.Project {
+// remove projects where user is not a collaborator
+func removeUnauthorizedProjects(cp []goship.Project, r *http.Request, u User) (fp []goship.Project) {
 	if authentication.authorization != true {
 		return cp
 	}
-	cleanProjects := []goship.Project{}
+
 	for _, p := range cp {
 		a := isCollaborator(p.RepoOwner, p.RepoName, u.UserName)
-		// If user does not have access to the project remove it.
 		if a == true {
-			cleanProjects = append(cleanProjects, p)
+			fp = append(fp, p)
+		}
+	}
+	return fp
+}
+
+//  set projects to lock where user is only in a pull only repo and append a comment
+func filterProject(p goship.Project, r *http.Request, u User) goship.Project {
+	if authentication.authorization != true {
+		return p
+	}
+
+	for i, e := range p.Environments {
+		// If the repo isn't already locked.. lock it if the user doesnt have permission
+		// and add to the comments
+		if e.IsLocked != true {
+			g := newGithubClient()
+			lock, err := userHasDeployPermission(g, p.RepoOwner, p.RepoName, u.UserName)
+			if err != nil {
+				log.Printf("Error getting Lock Permission: Locking anyway for safety %s", err)
+			}
+			p.Environments[i].IsLocked = !lock
+			// Add a line break if there is already a comment
+			if p.Environments[i].Comment != "" {
+				p.Environments[i].Comment = p.Environments[i].Comment + " | "
+			}
+			if !lock {
+				p.Environments[i].Comment = p.Environments[i].Comment + "you do not have permission to deploy "
+			}
+		} else {
```

Comment:
golang prefers less indent for normal flows.

probably the "else" clause is the more common flow in this function?
Then continue at the end of the "if" clause and dedent the "else" clause.